pkgs/development/compilers/crystal/default.nix

```diff
@@ -220,6 +220,8 @@ let
 
         install -Dm644 etc/completion.bash $out/share/bash-completion/completions/crystal
         install -Dm644 etc/completion.zsh $out/share/zsh/site-functions/_crystal
+        # fish completion was introduced since 1.6.0
+        test -f etc/completion.fish && install -Dm644 etc/completion.fish $out/share/fish/vendor_completions.d/crystal.fish
```

Comment:
use `installShellCompletion` for these.

Reply:
Simply replacing with `installShellCompletion` drops current files from `pkgs.crystal`. So added 
 `ln` after it. https://github.com/NixOS/nixpkgs/commit/7ad1c658a7057813b1058aa33c8e2d9f74604da8#diff-1205fbe1629d11896d8a36ee90fed97d79b6dbdf568bf4201483cdcf62a88318R230-R233